docs/source/getting_started.rst

```diff
@@ -0,0 +1,152 @@
+Getting started
+***************
+
+
+:class:`~torchsnapshot.Snapshot` is the core API of TorchSnapshot. The class represents application state persisted in storage. A user can take a snapshot of an application via :meth:`Snapshot.take() <torchsnapshot.Snapshot.take>` (a.k.a saving a checkpoint), or restore the state of an application from a snapshot via :meth:`Snapshot.restore() <torchsnapshot.Snapshot.restore>` (a.k.a loading a checkpoint).
+
+
+.. _app-state:
+
+
+Describing the application state
+--------------------------------
+
+Before using :class:`~torchsnapshot.Snapshot` to save or restore application state, the user needs to **describe the application state**. This is done by creating a dictionary that contains all **stateful objects** that the user wishes to capture as application state. Any object that exposes ``.state_dict()`` and ``.load_state_dict()`` are considered stateful objects. Common PyTorch objects such as :class:`~torch.nn.Module`, :class:`~torch.optim.Optimizer`, and LR schedulers all qualify as stateful objects and can be captured directly. Objects that don't meet this requirement can be captured via :class:`~torchsnapshot.StateDict`.
```

Comment:
do you want to mention and link the Stateful protocol here?

Reply:
I want to elaborate it in "API Reference" to make "Getting Started" more concise.